File: mobile/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final String? description;
  final double price;
  final double? discountPrice;
  final int stock;
  final String? imageUrl;
  final int? categoryId;
  final bool isActive;

  Product({
    required this.id,
    required this.name,
    this.description,
    required this.price,
    this.discountPrice,
    this.stock = 0,
    this.imageUrl,
    this.categoryId,
    this.isActive = true,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: json['price'].toDouble(),
      discountPrice: json['discount_price']?.toDouble(),
      stock: json['stock'] ?? 0,
      imageUrl: json['image_url'],
      categoryId: json['category_id'],
      isActive: json['is_active'] ?? true,
    );
  }

  double get effectivePrice => discountPrice ?? price;
  
  int get discountPercentage {
    if (discountPrice == null) return 0;
    return ((1 - discountPrice! / price) * 100).round();
  }
}

File: backend/app/schemas/product.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class CategoryCreate(BaseModel):
    name: str
    description: Optional[str] = None
    image_url: Optional[str] = None


class CategoryResponse(BaseModel):
    id: int
    name: str
    description: Optional[str]
    image_url: Optional[str]
    created_at: datetime

    class Config:
        from_attributes = True


class ProductCreate(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    discount_price: Optional[float] = None
    stock: int = 0
    image_url: Optional[str] = None
    category_id: Optional[int] = None


class ProductResponse(BaseModel):
    id: int
    name: str
    description: Optional[str]
    price: float
    discount_price: Optional[float]
    stock: int
    image_url: Optional[str]
    category_id: Optional[int]
    is_active: bool
    created_at: datetime

    class Config:
        from_attributes = True
